fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/trade/TradeOfferHelper.java

```diff
@@ -57,11 +57,9 @@ public static void registerWanderingTraderOffers(int level, Consumer<List<TradeO
 	}
 
 	/**
-	 * Refreshes the trade list by resetting the trade lists to vanilla state, and then registering all trade offers again.
-	 *
-	 * <p>This method is geared for use by mods which for example provide data driven villager trades.
+	 * @deprecated This never did anything useful.
 	 */
+	@Deprecated
 	public static void refreshOffers() {
```

Comment:
Not a fan of having methods that seeming do nothing. At the very least log a warning.

As for actaully making this I will need to find time to figure out what it was supposed to do

Reply:
Currently it's (expected to) remove from and then re-add all non-vanilla trade offers to the map.

Reply:
But that's entirely useless if the registered `Consumer<List<TradeOffers.Factory>>` are not invoked again, and the old impl doesn't do that.

Reply:
So either we make `refreshOffers` a no-op and let authors of data-driven trade mods remove their old factories from the map, or we introduce a breaking change by potentially calling the consumers multiple times.

Reply:
I would favor the former, with a warning in the log.

Reply:
After discussion with i5 on Discord, the latter might be better. His plan was that `refreshOffers` would remove all the trade offers registered by this API and then re-invoke all the registered consumers.

Technically this would be a subtle behavior change that could be considered breaking, but given how buggy this API is I think it will be fine.

Reply:
Ok, do we really care about `refreshOffers`? It's annoying to implement, and we can't really catch some edge cases where other mods could duplicate entries and/or reorder the underlying arrays. I'd just leave it tbh, if mods want to replace trade entries they can just remove (it's a public map so it doesn't even require mixin, and we can add support later if necessary) and re-add trades.

Reply:
> His plan was that `refreshOffers` would remove all the trade offers registered by this API and then re-invoke all the registered consumers.

Isn't this exactly my original suggestion :thinking: Imo you should just leave this no-op and log liek what you currently have.

In addition, for resource reloading, I personally think that mod can implement reload-aware trade offer factories than relying on this refresh to update things.

Reply:
Updated with the log, please let me know if I should un-resolve this conversation.